### PR TITLE
Reduce Concurrent Tests

### PR DESCRIPTION
**Pull request by anissa111**
Addresses #473

Summary of changes:
- adds separate `ci-nightly.yml` workflow that runs on all `os` and `python-version`s at 1 AM nightly
- modifies `ci.yml` to cancel previous workflows and to run `macos-latest` on oldest and newest versions of python we support (3.7 and 3.10) to help concurrency limit (all versions run nightly)

### CODE REVIEW

**Comment by anissa111** [comment]
Note that previous tests on a branch can now be automatically cancelled as shown last commit

**Review by michaelavs — APPROVED** [review]
Approving as it looks like all checks will pass with these changes (I assume the sphinx one would/will pass if given enough time)

**Comment by anissa111** [comment]
That... actually shouldn't be there. Huh. I wonder if it's because it's not in main? If you click on the actions for the PR, it shows them all as passing. This PR specifically removed macos-latest 3.9

**Comment by hCraker** [comment]
> Approving as it looks like all checks will pass with these changes (I assume the sphinx one would/will pass if given enough time)

@michaelavs Do you think we should wait for it to pass before merging, or is it worth risking? I highly doubt a preemptive merge will break everything (I could regret saying that)

**Comment by anissa111** [comment]
@hCraker We don't have the permissions to do that anymore haha

**Comment by hCraker** [comment]
> @hCraker We don't have the permissions to do that anymore haha

Oh darn I didn't even notice we lost that since I used it so rarely

**Comment by anissa111** [comment]
@erogluorhan We think this might need your admin privileges. The checks on the PR start with just those we've made in the PR and then adds at least one extra expected required test (from main?) that is one of the tests that we've removed in this PR and on it indefinitely, which blocks merging.

Here is a screenshot of the tests right after I closed and reopened the PR, which lists the expected number of tests
image

Here is a screenshot taken a few minutes afterward that shows that two extra tests have been added. I think one is `docs/readthedocs.org:geocat-examples` and the other is `macos-latest 3.9`
image1

I am unclear why only `macos-latest 3.9` is acting like this and not `macos-latest 3.8`, which was also removed in this PR, but I suspect it is because 3.8 tests were added relatively recently and might not be part of the branch protection rules for main, but we don't have the permissions to see those, either.

If you look at the actions tab, it shows [all tests pass](https://github.com/NCAR/GeoCAT-examples/actions/runs/2678655490) and lists the expected combinations of os and python version.

When you have a minute, could you check this out and maybe lend this PR your magic admin privileges?

**Comment by erogluorhan** [comment]
> > Approving as it looks like all checks will pass with these changes (I assume the sphinx one would/will pass if given enough time)
> 
> @michaelavs Do you think we should wait for it to pass before merging, or is it worth risking? I highly doubt a preemptive merge will break everything (I could regret saying that)

It should (almost) never be worth risking the `main` branch without seeing all of them pass.

**Comment by erogluorhan** [comment]
> @hCraker We don't have the permissions to do that anymore haha

As soon as we have package leaders listed, let us give them the admin privileges, too (but, with they swearing not to risk the `main` branch in any circumstances haha)

**Comment by erogluorhan** [comment]
@anissa111 sorry if I am too late, but I figured why only 3.9 was acting like that. Here is what the current state of our branch protection rule was (3.8 was not listed there):

image

**Comment by anissa111** [comment]
@erogluorhan No problem! I figured it was something like that. I suggest removing `macos-latest 3.9` and adding `macos-latest 3.10`, `ubuntu-latest 3.8`, and `ubuntu-latest 3.10`

**Comment by anissa111** [comment]
@erogluorhan Great, thank you!!

File: .github/workflows/ci.yml
Comment (erogluorhan) [inline .github/workflows/ci.yml:21]: Instead of removing macos-latest at all, I'd do ubuntu and macos but only for versions 3.7 and 3.10 for this pull-request CI

File: .github/workflows/ci-nightly.yml
Comment (erogluorhan) [inline .github/workflows/ci-nightly.yml:1]: I'd name this CI Nightly

Comment (anissa111) [reply .github/workflows/ci-nightly.yml:1]: I agree! See https://github.com/NCAR/GeoCAT-examples/pull/475

**Pull request merged**
